code/datums/components/omen.dm

```diff
@@ -43,14 +53,41 @@
 /datum/component/omen/proc/check_accident(atom/movable/our_guy)
 	SIGNAL_HANDLER_DOES_SLEEP
 
+	if(!isliving(our_guy))
+		return
+
+	var/mob/living/living_guy = our_guy
+
 	if(!prob(15))
 		return
-	for(var/t in get_adjacent_open_turfs(our_guy))
+	var/our_guy_pos = get_turf(living_guy)
+	for(var/turf_content in our_guy_pos)
+		if(istype(turf_content, /obj/machinery/door/airlock))
+			to_chat(living_guy, "<span class='warning'>A malevolent force launches your body to the floor...</span>")
+			var/obj/machinery/door/airlock/darth_airlock = turf_content
+			living_guy.apply_status_effect(STATUS_EFFECT_KNOCKDOWN, 10, TRUE)
+			living_guy.apply_status_effect(STATUS_EFFECT_STUN, 10, TRUE)
```

Comment:
Why don't you add a "wormed" new status effect for this, with both the effects of knockdown and stun?

Reply:
Wormed status effect is now implemented!